rest/accounts/v1/credentials_aws.go

```diff
@@ -152,6 +154,75 @@ func (c *ApiService) ListCredentialAws(params *ListCredentialAwsParams) (*ListCr
 	return ps, err
 }
 
+//Retrieve a single page of  records from the API. Request is executed immediately.
+func (c *ApiService) CredentialsAWSPage(params *ListCredentialAwsParams, pageToken string, pageNumber string) (*ListCredentialAwsResponse, error) {
+	path := "/v1/Credentials/AWS"
+
+	data := url.Values{}
+	headers := make(map[string]interface{})
+
+	if params != nil && params.PageSize != nil {
+		data.Set("PageSize", fmt.Sprint(*params.PageSize))
+	}
+
+	data.Set("PageToken", pageToken)
+	data.Set("PageNumber", pageNumber)
+
+	resp, err := c.requestHandler.Get(c.baseURL+path, data, headers)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	ps := &ListCredentialAwsResponse{}
+	if err := json.NewDecoder(resp.Body).Decode(ps); err != nil {
+		return nil, err
+	}
+
+	return ps, err
+}
+
+//Lists CredentialsAWS records from the API as a list. Unlike stream, this operation is eager and will loads 'limit' records into memory before returning.
+func (c *ApiService) CredentialsAWSList(params *ListCredentialAwsParams, limit int) ([]ListCredentialAwsResponse, error) {
+	params.SetPageSize(c.requestHandler.ReadLimits(params.PageSize, limit))
+	response, err := c.ListCredentialAws(params)
+	if err != nil {
+		return nil, err
+	}
+
+	page := client.NewPage(c.baseURL, response)
+
+	resp := c.requestHandler.List(page, limit, 0)
+	ret := make([]ListCredentialAwsResponse, len(resp))
+
+	for i := range resp {
+		jsonStr, _ := json.Marshal(resp[i])
+		ps := ListCredentialAwsResponse{}
+		if err := json.Unmarshal(jsonStr, &ps); err != nil {
```

Comment:
What's the difference between this and `json.NewDecoder().Decode`?

Reply:
`json.NewDecoder().Decode` takes in the data from an `io.Reader` byte stream whereas `json.Unmarshal` takes in the json data as a string.